### ISSUE TITLE
Stack overflow in source generator?

### ISSUE THREAD

**Issue by DamianEdwards** [DapperLib/DapperAOT#94]
I tried to setup one of the ASP.NET Core benchmarks app's ([this one](https://github.com/aspnet/Benchmarks/tree/damianedwards/dapper.aot/src/BenchmarksApps/TodosApi)) to use Dapper.AOT but it appears to be causing a stack overflow with lots of the following printed to output during build:

```
C:\Program Files\dotnet\sdk\8.0.100\Roslyn\Microsoft.CSharp.Core.targets(84,5): error :    at Dapper.Internal.Inspection.InvolvesTup
leType(Microsoft.CodeAnalysis.ITypeSymbol, Boolean ByRef) [D:\src\GitHub\aspnet\Benchmarks\src\BenchmarksApps\TodosApi\TodosApi.cspr
oj]
```

Visual Studio freezes and then disappears.

- Dapper version 2.0.123
- Dapper.AOT version 1.0.23
- Npgsql

**To Reproduce**
Project that I was updating is: https://github.com/aspnet/Benchmarks/tree/damianedwards/dapper.aot/src/BenchmarksApps/TodosApi

**Expected behavior**
Expect it to work.

**Screenshots**
See error above.

**Additional context**
none

**Comment by mgravell** [comment]
A quick attempt to repro was unsuccessful; I *suspect* that this is a nuanced thing that involves your migration and wrapper code from `DbDataSource` (which isn't directly supported in Dapper until vNext). Any chance you have an intermediate branch with your changes that show this behaviour?

**Comment by DamianEdwards** [comment]
The link is to the intermediate branch with these changes 😁
Let me try to repro on a different machine.

**Comment by DamianEdwards** [comment]
I just repro'ed this on another machine using the branch at https://github.com/aspnet/Benchmarks/tree/damianedwards/dapper.aot/

Clone that, then navigate to `./src/BenchmarksApps/TodosApi` and run `dotnet build`.

**Comment by mgravell** [comment]
OK, this was a PITA to investigate - it seems to be fixed in 1.0.27, but getting a proper reliable repro was elusive, even using a local nuget folder and proper restore; however, it is *either* something to do with error-types or unbound generics; I've fixed both of those scenarios; I've also added a general depth check, and better fault storage

yikes, that was ... fun

for small values of "fun"
